install/kubernetes/cilium/templates/spire/agent/daemonset.yaml

```diff
@@ -86,6 +86,8 @@ spec:
       {{- with .Values.authentication.mutual.spire.install.agent.tolerations }}
       tolerations:
         {{- toYaml . | trim | nindent 8 }}
+        - key: {{ .Values.agentNotReadyTaintKey }}
```

Comment:
does this needs to be conditional? because it's only in the agent config if not empty -> https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-configmap.yaml#L1181-L1183

Reply:
looks like if not set, the Cilium Agent defaults to `node.cilium.io/agent-not-ready` (https://github.com/cilium/cilium/blob/main/pkg/defaults/node.go#L45) 

at best we default to this here in the helm chart if `.Values.agentNotReadyTaintKey` is set to empty - otherwise they won't match. :thinking:

Reply:
we can avoid the if condition and set the default with the respective helm function

```
        - key: {{ .Values.agentNotReadyTaintKey | default "node.cilium.io/agent-not-ready" }}
          effect: NoSchedule
```